lib/src/html/html_generator_instance.dart

```diff
@@ -413,6 +415,7 @@ class HtmlGeneratorInstance {
     String outFile = path.joinAll(filename.split('/'));
     String content = template.renderString(data);
 
+    content = content.replaceAll(HTMLBASE_PLACEHOLDER, data.htmlBase);
```

Comment:
I do think it's possible to not do this, but we'd have to really lock down TemplateData.  Could you add a TODO to get rid of this kludge?

Reply:
I left a large comment over the placeholder string, it includes some context, a link to #2090, and a `TODO`.